server/server.go

```diff
@@ -67,21 +67,16 @@ func (s *mmoServer) start(protocol string, port int, errc chan error) error {
 				w.WriteHeader(http.StatusNoContent)
 				return
 			}
-			for client, checksum := range clientChecksums {
-				if strings.Contains(req.URL.Path, client) && req.URL.Query().Get("checksum") == checksum {
-					w.WriteHeader(http.StatusNoContent)
-					return
-				}
-				if req.URL.Path == "/"+client {
```

Comment:
e.g. `/files/client` instead of `/client` will break this. so we should just check for suffix